phat/contracts/action_offchain_rollup/lib.rs

```diff
@@ -215,89 +214,66 @@ mod action_offchain_rollup {
+                &data,
+                &KeyPair::from(self.attest_key),
+            )
+            .log_err("failed to sign transaction")
+            .or(Err(Error::FailedToSignTransaction))?;
+            Ok(ethabi::encode(&[tx, Token::Bytes(sig.0)]))
+        }
 
-            let headers = vec![
-                ("Content-Type".into(), "application/json".into()),
-                ("User-Agent".into(), "phat-contract".into()),
-            ];
-            let body = format!("{{\"query\":\"\\n          query Profile {{\\n            profile(request: {{ profileId: \\\"{profile_id}\\\" }}) {{\\n              stats {{\\n                totalFollowers\\n                totalFollowing\\n                totalPosts\\n                totalComments\\n                totalMirrors\\n                totalPublications\\n                totalCollects\\n              }}\\n            }}\\n          }}\\n          \"}}");
-
-            let resp = pink::http_post!(lens_api, body.as_bytes(), headers);
-            if resp.status_code != 200 {
-                pink::warn!(
-                    "Fail to read Lens api withstatus code: {}, reason: {}, body: {:?}",
-                    resp.status_code,
-                    resp.reason_phrase,
-                    resp.body
-                );
-                return Err(Error::FailedToFetchData);
-            }
 
-            let resp_body = String::from_utf8(resp.body).or(Err(Error::FailedToDecode))?;
-            Ok(resp_body)
+        /// Processes a request with the the core js and returns the output without signature.
+        #[ink(message)]
+        pub fn get_raw_answer(&self, request: Vec<u8>) -> Result<(Vec<u8>, CodeHash)> {
+            self.handle_request(&request)
         }
 
-        pub fn transform_data(data: String, transform_js: String) -> Result<u128> {
-            let result = js::eval(transform_js.as_str(), &[data])
-                .map_err(|_| Error::FailedToTransformData)?;
-
-            let result_num: u128 = match result {
-                phat_js::Output::String(result_str) => {
-                    result_str.parse().map_err(|_| Error::BadTransformedData)?
-                }
-                phat_js::Output::Bytes(_) => {
-                    return Err(Error::BadTransformedData);
-                }
+        /// Processes a request with the the core js and returns the output.
+        fn handle_request(&self, request: &[u8]) -> Result<(Vec<u8>, CodeHash)> {
+            let Some((core_js, js_hash)) = &self.core_js else {
+                pink::error!("HandlerNotConfigured");
+                return Err(Error::HandlerNotConfigured);
             };
-            Ok(result_num)
+            let mut args = alloc::vec![alloc::format!("0x{}", hex_fmt::HexFmt(request))];
+            if let Some(settings) = &self.core_settings {
+                args.push(settings.clone());
+            }
+            let output = js::eval(core_js, &args).map_err(Error::JsError)?;
```

Comment:
What if query timeout happens here? Can we give a valid error to users to help them find it out?

Reply:
It depends on the core js implementation. The js can catch the timeout error of the http request using the new API batchHttpRequest and rethrow it. In this way caller can check the error description in the JsError.
Note: The API httpRequest doesn't support timeout.
Let me make it as an example.